ui/app/src/components/DatasourceList/DatasourceEditorForm.tsx

```diff
@@ -0,0 +1,228 @@
+  const handleCancel = useCallback(() => {
+    if (JSON.stringify(initialState) !== JSON.stringify(state)) {
+      setDiscardDialogStateOpened(true);
+    } else {
+      onClose();
+    }
+  }, [state, initialState, setDiscardDialogStateOpened, onClose]);
+
+  return (
+    <>
+      <Box
+        sx={{
+          display: 'flex',
+          alignItems: 'center',
+          padding: (theme) => theme.spacing(1, 2),
+          borderBottom: (theme) => `1px solid ${theme.palette.divider}`,
+        }}
+      >
+        <Typography variant="h2">{saveActionStr} Datasource</Typography>
+        <Stack direction="row" spacing={1} sx={{ marginLeft: 'auto' }}>
+          <Button disabled={isReadonly || !validation.isValid} variant="contained" onClick={handleSave}>
+            {saveActionStr}
+          </Button>
+          <Button color="secondary" variant="outlined" onClick={handleCancel}>
+            Cancel
+          </Button>
+          {onDelete && (
+            <Button disabled={isReadonly} color="error" variant="outlined" onClick={onDelete}>
+              Delete
+            </Button>
+          )}
+        </Stack>
+      </Box>
+      <Box padding={2} sx={{ overflowY: 'scroll' }}>
+        <Grid container spacing={2} mb={2}>
+          <Grid item xs={4}>
+            <TextField
+              required
+              error={!!validation.name}
+              fullWidth
+              label="Name"
+              value={state.metadata.name}
+              InputProps={{
+                readOnly: isReadonly,
+              }}
+              helperText={validation.name}
+              onChange={(v) => {
+                setState((draft) => {
+                  draft.metadata.name = v.target.value;
+                });
+              }}
+            />
+          </Grid>
+          <Grid item xs={8}>
+            <TextField
+              fullWidth
+              label="Display Label"
+              value={state.spec.display?.name}
+              InputProps={{
+                readOnly: isReadonly,
+              }}
+              onChange={(v) => {
+                setState((draft) => {
+                  if (draft.spec.display) {
+                    draft.spec.display.name = v.target.value;
+                  }
+                });
+              }}
+            />
+          </Grid>
+          <Grid item xs={12}>
+            <TextField
+              fullWidth
+              label="Description"
+              value={state.spec.display?.description}
+              InputProps={{
+                readOnly: isReadonly,
+              }}
+              onChange={(v) => {
+                setState((draft) => {
+                  if (draft.spec.display) {
+                    draft.spec.display.description = v.target.value;
+                  }
+                });
+              }}
+            />
+          </Grid>
+          <Grid item xs={6} sx={{ paddingTop: '5px !important' }}>
```

Comment:
'would have like avoiding using !important here but didn't find an easy way... it seems quite some people already complained about this https://github.com/mui/material-ui/issues/8509